src/common/sharedlibrary/CMakeLists.txt

```diff
@@ -14,6 +14,24 @@
 #
 # Authored by: Alexandros Frantzis <alexandros.frantzis@canonical.com>
 
+include(CheckFunctionExists)
+
+list(APPEND CMAKE_REQUIRED_LIBRARIES dl)
+list(APPEND CMAKE_REQUIRED_DEFINITIONS -D_GNU_SOURCE)
+
+# CMake docs suggests that we should, instead, be using check_symbol_exists.
+# The only problem there is that check_symbol_exists("dlvsym", "dlfcn.h", HAS_DLVSYM) never finds
+# the dlvsym symbol ☹.
+check_function_exists("dlvsym" HAS_DLVSYM)
+
+if (NOT HAS_DLVSYM)
+  add_compile_definitions(DONT_USE_DLVSYM)
```

Comment:
Nits:

1. We should namespace our #defines with "MIR_"
2. This is only needed by one source file so prefer `set_source_files_properties()`